lighthouse-core/gather/driver.js

```diff
@@ -743,6 +743,20 @@ class Driver {
   }
 
   /**
+   * Returns the flattened list of all DOM nodes within the document.
+   * @param {boolean=} pierce Whether to pierce through shadow trees and iframes.
+   *     True by default.
+   * @return {!Promise<!Array<!Element>>} The found elements, or [], resolved in a promise
+   */
+  getAllElementsInDocumentAndFrames(pierce = true) {
```

Comment:
right, but `getAllElementsInDocumentAndFrames(false)` is confusing on its own. Don't get all elements?

We could change it somehow to avoid the boolean trap, but I was just suggesting that we don't need to because we don't actually have any uses for not piercing.

If you really think it should stay, though, we should rename somehow (if `pierce` is false it won't be looking in the "frames" part of `getAllElementsInDocumentAndFrames`), and maybe make the parameter `noPierce` so that `undefined` can just act as false as is usually expected?